types/utils.go

```diff
@@ -40,39 +38,53 @@ func GetRandomValidatorSetWithPrivKey() (*cmtypes.ValidatorSet, ed25519.PrivKey)
 	return valset, privKey
 }
 
-// GetValidatorSet returns a validator set with a single validator
-// with the given key
-func GetValidatorSet(privKey ed25519.PrivKey) *cmtypes.ValidatorSet {
-	pubKey := privKey.PubKey()
-	valset := cmtypes.NewValidatorSet(
-		[]*cmtypes.Validator{
-			{PubKey: pubKey, Address: pubKey.Address(), VotingPower: 1},
-		},
-	)
-	return valset
+// BlockContext carries all necessary state for block generation, including configurations and outputs.
+type BlockContext struct {
+	Height       uint64
+	NTxs         int
+	PrivKey      ed25519.PrivKey // Input and Output option
+	ProposerAddr []byte          // Input option
+	Block        *Block          // Output: the generated block.
+	// Add other fields as necessary for inputs and outputs.
 }
 
-// GetRandomBlock returns a block with random data
+// GetRandomBlock generates a block with specified height and number of transactions.
+// It's tailored for simplicity, primarily used in test setups where additional outputs are not needed.
 func GetRandomBlock(height uint64, nTxs int) *Block {
-	block, _ := GetRandomBlockWithKey(height, nTxs, nil)
-	return block
+	ctx := BlockContext{
+		Height:       height,
+		NTxs:         nTxs,
+		PrivKey:      nil,
+		ProposerAddr: nil,
+	}
+	// Assuming GenerateBlock modifies the context directly with the generated block and other needed data.
+	err := GenerateRandomBlockWithContext(&ctx) // Simplified: Error handling is skipped for brevity.
+	if err != nil {
+		panic(err)
+	}
+
+	return ctx.Block
 }
 
-// GetRandomBlockWithKey returns a block with random data and a signing key
-func GetRandomBlockWithKey(height uint64, nTxs int, privKey ed25519.PrivKey) (*Block, ed25519.PrivKey) {
-	block := getBlockDataWith(nTxs)
+// GetRandomBlock returns a block with random data and the given height
+func GenerateRandomBlockWithContext(ctx *BlockContext) error {
```

Comment:
let's revert this error as well and use panics again.

Also let's call this `GenerateRandomBlockCustrom` instead. This will align with the updated naming of the type better as well. 

This also supports the structure of:
```go
func GetXYZ() {
  return GetXYZCustom(defaults)
}

func GetXYZCustom(defaults Options)
```

Please update the doc string as well.